### PR TITLE
Readonly property

### PR DESCRIPTION
**Pull request by psychon**
Some readonly properties and getting rid of useless no-op setters.

### CODE REVIEW

File: tests/test-miss-handlers.lua [hunk 1]
```diff
@@ -40,4 +40,8 @@ local w = wibox()
 w.foo = "bar"
 assert(w.foo == "bar")
 
+-- Test if read-only properties really are read-only
+screen[1].clients = 42
+assert(screen[1].clients ~= 42)
```
Comment (blueyed) [inline tests/test-miss-handlers.lua]: Should there be a warning or even error in this case?
Something like `cannot set read-only attribute: clients`?

Comment (psychon) [inline tests/test-miss-handlers.lua]: I could do that in the other cases, but in this case the C code "eats" all `error()`s and turns them into warnings. Should I do this anyway?

Comment (blueyed) [inline tests/test-miss-handlers.lua]: Thanks!

Comment (psychon) [inline tests/test-miss-handlers.lua]: I guess that means "please squash the fixup commits into the related commits". Ok. :-)

**Comment by blueyed** [comment]
Yeah, please squash and then merge.

**Comment by psychon** [comment]
Done. However, I first pushed awesome master and then the PR for this branch and now github turned this into the empty pull request... Whatever.

**Pull request closed without merging**